File: Backend/store/src/main/kotlin/com/expediagroup/graphql/demo/models/CustomProduct.kt
package com.expediagroup.graphql.demo.models

import javax.persistence.*
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull

@Entity
data class CustomProduct(
  @OneToOne()
  @JoinColumn(name = "user_id")
  var user: User,

  @get: NotNull
  var max_price: Float,
  @get: NotNull
  var min_price: Float,

  @get: NotNull
  var max_days: Int,
  @get: NotNull
  var min_days: Int,

  @get: NotBlank
  override var description: String,

  @get: NotNull
  var size: Float = 0f
) : Product(){
  init{
    name = "CustomProduct"
  }
  var req_status: ReqStatus = ReqStatus.REVIEW_PENDING
  var req_text: String? = ""
}


enum class ReqStatus {
  REVIEW_PENDING, IN_REVIEW, ANSWER
}

File: src/api/customProduct.js
import { gql } from '@apollo/client';

const ADD_PRODUCT = gql`
  mutation AddCustomProduct(
    $userId: Long!, $max_days: Int!, $min_days: Int!,
    $min_price: Float!, $max_price: Float!, $description: String!, $size: Float!
  ) {
    addCustomProduct(
      userId: $userId
      max_price: $max_price
      min_price: $min_price
      min_days: $min_days
      max_days: $max_days
      description: $description
      size: $size
    ) {
      id
    }
  }
`;

const GET_PRODUCTS = gql`
  query GetCustomProducts($userId: Long!, $page: Long) {
    customProducts(
      userId: $userId,
      page: $page,
    ) {
      id
      description
      size
      max_price
      min_price
      max_days
      min_days
      req_status
      req_text
    }
  }
`;

const GET_PRODUCT = gql`
  query GetCustomProduct($id: Long!) {
    customProduct(
      id: $id
    ) {
      id
      name
      description
      size
      max_price
      min_price
      max_days
      min_days
      req_status
      req_text
    }
  }
`;

export {
  GET_PRODUCTS,
  GET_PRODUCT,
  ADD_PRODUCT
}
